client/cli/src/config.rs

```diff
@@ -0,0 +1,315 @@
+		Ok(Some(Vec::new()))
+	}
+
+	/// Get the prometheus port (`None` if disabled)
+	fn get_prometheus_port(&self) -> Result<Option<SocketAddr>> {
+		Ok(Default::default())
+	}
+
+	/// Get the telemetry endpoints (if any)
+	fn get_telemetry_endpoints<G, E>(
+		&self,
+		chain_spec: &ChainSpec<G, E>,
+	) -> Result<Option<TelemetryEndpoints>> {
+		Ok(chain_spec.telemetry_endpoints().clone())
+	}
+
+	/// Get the telemetry external transport
+	fn get_telemetry_external_transport(&self) -> Result<Option<ExtTransport>> {
+		Ok(Default::default())
+	}
+
+	/// Get the default value for heap pages
+	fn get_default_heap_pages(&self) -> Result<Option<u64>> {
+		Ok(Default::default())
+	}
+
+	/// Returns `Ok(true)` if offchain worker should be used
+	fn get_offchain_worker(&self, _roles: Roles) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Get sentry mode (i.e. act as an authority but **never** actively participate)
+	fn get_sentry_mode(&self) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Returns `Ok(true)` if authoring should be forced
+	fn get_force_authoring(&self) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Returns `Ok(true)` if grandpa should be disabled
+	fn get_disable_grandpa(&self) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Get the development key seed from the current object
+	fn get_dev_key_seed(&self, _is_dev: bool) -> Result<Option<String>> {
+		Ok(Default::default())
+	}
+
+	/// Get the tracing targets from the current object (if any)
+	fn get_tracing_targets(&self) -> Result<Option<String>> {
+		Ok(Default::default())
+	}
+
+	/// Get the TracingReceiver value from the current object
+	fn get_tracing_receiver(&self) -> Result<sc_tracing::TracingReceiver> {
+		Ok(Default::default())
+	}
+
+	/// Get the node key from the current object
+	fn get_node_key(&self, _net_config_dir: &PathBuf) -> Result<NodeKeyConfig> {
+		Ok(Default::default())
+	}
+
+	/// Create a Configuration object from the current object
+	fn create_configuration<C: SubstrateCLI<G, E>, G, E>(
+		&self,
+		task_executor: Arc<dyn Fn(Pin<Box<dyn Future<Output = ()> + Send>>) + Send + Sync>,
+	) -> Result<Configuration<G, E>>
+	where
+		G: RuntimeGenesis,
+		E: ChainSpecExtension,
+	{
+		let chain_spec = self.get_chain_spec::<C, G, E>()?;
+		let is_dev = self.get_is_dev()?;
+		let default_config_dir = app_dirs::get_app_root(
+			AppDataType::UserData,
+			&AppInfo {
+				name: C::get_executable_name(),
+				author: C::get_author(),
+			},
+		)
+		.expect("app directories exist on all supported platforms; qed");
+		let config_dir = self
+			.get_base_path()?
+			.unwrap_or(&default_config_dir)
+			.join("chains")
+			.join(chain_spec.id());
+		let net_config_dir = config_dir.join(DEFAULT_NETWORK_CONFIG_PATH);
+		let client_id = C::client_id();
+		// TODO: this parameter is really optional, shouldn't we leave it to None?
+		let database_cache_size = Some(self.get_database_cache_size()?.unwrap_or(1024));
```

Comment:
@bkchr this is a parameter you set to 1024 in 2b716b5a96e2806a94200434c0f54a10114a126c

Should I:
- keep it that way
- move the default value of 1024 to sc_service (as suggested by @tomaka )
- delete it and leave it to none

Reply:
Correction: this value has been changed to 32 later on c3ff85d13b8b660c2c4eb850209c06a9397bed3e